homeassistant/components/homekit/type_switches.py

```diff
@@ -226,3 +233,54 @@ def async_update_state(self, new_state):
         self.char_active.set_value(current_state)
         _LOGGER.debug("%s: Set in_use state to %s", self.entity_id, current_state)
         self.char_in_use.set_value(current_state)
+
+
+@TYPES.register("SelectSwitch")
+class SelectSwitch(HomeAccessory):
+    """Generate a Switch accessory that contains multiple switches."""
+
+    def __init__(self, *args):
+        """Initialize a Switch accessory object."""
+        super().__init__(*args, category=CATEGORY_SWITCH)
+
+        state = self.hass.states.get(self.entity_id)
+        self.select_chars = {}
+        options = state.attributes.get("options", [])
+        if options:
+            for option in options:
+                serv_option = self.add_preload_service(
+                    SERV_OUTLET, [CHAR_NAME, CHAR_IN_USE]
+                )
+                serv_option.configure_char(
+                    CHAR_NAME,
+                    value=f"{option}"[:MAX_NAME_LENGTH],
+                )
+                serv_option.configure_char(CHAR_IN_USE, value=False)
+                self.select_chars[option] = serv_option.configure_char(
+                    CHAR_ON,
+                    value=False,
+                    setter_callback=lambda value, option=option: self.select_option(
+                        option
+                    ),
+                )
+            self.set_primary_service(self.select_chars[options[0]])
+        # Set the state so it is in sync on initial
+        # GET to avoid an event storm after homekit startup
+        self.async_update_state(state)
+
+    def select_option(self, option):
+        """Set preset_mode if call came from HomeKit."""
+        _LOGGER.debug("%s: Set option to %d", self.entity_id, option)
+        params = {ATTR_ENTITY_ID: self.entity_id}
+        if option:
+            params["option"] = option
+            self.async_call_service(INPUT_SELECT_DOMAIN, SERVICE_SELECT_OPTION, params)
+
+    @callback
+    def async_update_state(self, new_state):
+        """Update switch state after state changed."""
+        current_option = new_state.state
+        for option, char in self.select_chars.items():
+            hk_value = 1 if option == current_option else 0
+            if char.value != hk_value:
+                char.set_value(hk_value)
```

Comment:
```suggestion
            char.set_value(hk_value)
```

You can drop the guard as its not needed with newer `HAP-python`